Cutelyst/application.cpp

```diff
@@ -441,6 +442,96 @@ bool Application::enginePostFork()
     return true;
 }
 
+void Application::addTranslator(const QLocale &locale, QTranslator *translator)
+{
+    Q_D(Application);
+    Q_ASSERT_X(translator, "add translator to application", "invalid QTranslator object");
+    if (d->translators.contains(locale)) {
+        QVector<QTranslator*> trs = d->translators.value(locale);
+        trs.prepend(translator);
+        d->translators.insert(locale, trs);
+    } else {
+        d->translators.insert(locale, QVector<QTranslator*>(1, translator));
+    }
+}
+
+void Application::addTranslator(const QString &locale, QTranslator *translator)
+{
+    addTranslator(QLocale(locale), translator);
+}
+
+void Application::addTranslators(const QLocale &locale, const QVector<QTranslator *> &translators)
+{
+    Q_D(Application);
+    Q_ASSERT_X(!translators.empty(), "add translators to application", "empty translators vector");
+    if (d->translators.contains(locale)) {
```

Comment:
Also with an iterator that isn't const you could avoid inserting the item on the hash again manipulating the value directly, which avoids a temporary QVector and detaching.